src/connection.rs

```diff
@@ -695,8 +720,10 @@ impl<Exe: Executor> Connection<Exe> {
 
         let hostname = hostname.unwrap_or_else(|| address.ip().to_string());
 
-        debug!("Connecting to {}: {}", url, address);
+        let id = rand::random();
```

Comment:
I think rand::random is fine since there wouldn't be a hugh chance of generating same number but it would be idea if we could leave a comment for possbile random number conflict!

Reply:
Let's use UUID